examples/acl/src/backend/app/controllers/posts.clj

```diff
@@ -0,0 +1,193 @@
+  "Basic SQL queries by request-method"
+  {:get    (-> (select :*)
+               (from :posts))
+   :post   (helpers/update :posts)
+   :put    (insert-into :posts)
+   :delete (delete-from :posts)})
+
+(defn base-query
+  "Inserting SQL query into state"
+  [{{method :request-method} :request :as state}]
+  (xiana/ok (assoc state :query (query-map method))))
+
+(defn handle-id-map
+  [query id method]
+  (when id (let [uuid-id (UUID/fromString id)]
+             (get {:get    (-> query (where [:= :id uuid-id]))
+                   :post   (-> query (where [:= :id uuid-id]))
+                   :delete (-> query (where [:= :id uuid-id]))}
+                  method))))
+
+(defn handle-id
+  "Add where clause to SQL query if it's necessary"
+  [{{{id :id} :query-params
+     method   :request-method} :request
+    query                      :query
+    :as                        state}]
+  (if-let [new-query (handle-id-map query id method)]
+    (xiana/ok (assoc state :query new-query))
+    (xiana/ok state)))
+
+(defn view
+  "Prepare and run view"
+  [{view :view :as state}]
+  (view state))
+
+(defn restriction-map
+  [query user-id case-v]
+  (get {[:own :get]    (-> query (merge-where [:= :user_id user-id]))
+        [:own :post]   (-> query (merge-where [:= :user_id user-id]))
+        [:own :delete] (-> query (merge-where [:= :user_id user-id]))}
+       case-v))
+
+(defn restriction
+  "Extends WHERE clause if necessary, to check data-ownership"
+  [{{restriction :acl}                               :response-data
+    query                                            :query
+    {{{{user-id :id} :user} :session-data} :session} :deps
+    {method :request-method}                         :request
+    :as                                              state}]
+  (if-let [new-query (restriction-map query user-id [restriction method])]
+    (xiana/ok (assoc state :query new-query))
+    (xiana/ok state)))
+
+(defn db-call
+  "Runs SQL query, inserts result into state"
+  [{query :query :as state}]
+  (let [result (execute state (sql/format query))]
+    (xiana/ok (assoc-in state [:response-data :db-data] result))))
+
+(defn handle-body-map
+  [query form-params user-id method]
+  (let [content (:content form-params)]
+    (get {:post (-> query (sset {:content content}))
+          :put  (-> query (columns :content :user_id) (values [[content user-id]]))}
+         method)))
+
+(defn handle-body
+  "Add content from form params"
+  [{{form-params :form-params
+     method      :request-method}                    :request
+    {{{{user-id :id} :user} :session-data} :session} :deps
+    query                                            :query
+    :as                                              state}]
+  (if-let [new-query (handle-body-map query form-params user-id method)]
+    (xiana/ok (assoc state :query new-query))
+    (xiana/ok state)))
+
+(defn controller
+  "Example controller for ACL, and DataOwnership"
+  [state]
+  (xiana/flow->
```

Comment:
Assume interceptors for auth, session, ACL will happen in pre-controller context.
For the rest, use more descriptive names, so it is easier to understand. Like
```
(xiana/flow-> 
    state 
    build-query 
    execute-query 
    view)
```